sqlite_test.go

```diff
@@ -1072,3 +1072,79 @@ func BenchmarkBeginTxNoop(b *testing.B) {
 // TODO(crawshaw): test TextMarshaler
 // TODO(crawshaw): test named types
 // TODO(crawshaw): check coverage
+
+// This tests for https://github.com/tailscale/sqlite/issues/73
+func TestPrepareReuse(t *testing.T) {
+	db := openTestDB(t)
+	ctx := context.Background()
+	sqlConn, err := db.Conn(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer sqlConn.Close()
+
+	// Insert a bunch of values into a table that we'll query to get
+	// multiple rows back.
+	err = ExecScript(sqlConn,
+		`BEGIN;
+		CREATE TABLE t (c);
+		INSERT INTO t VALUES (1), (2), (3), (4);
+		COMMIT;`)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	ctx = WithPersist(ctx)
+
+	// Calling PrepareContext twice in a row used to return the same
+	// statement to both callers.
+	const query = "SELECT c FROM t;"
```

Comment:
FROM t ORDER BY c

otherwise it's not defined the order they're returned, which your tests depends on below